### ISSUE TITLE
Add opt-in preference for "Enter reader view automatically"

### ISSUE THREAD

**Issue by mthurmond** [minbrowser/min#760]
*Problem:*
I love reader view and always want to use it when it's available for a page. However, I forget to click the button most times and, when I do remember, it's a long micro-interaction to have to scroll to the top and click a small button for each content page I visit. 

*Request:*
In preferences, give me an option to "Enter reader view automatically". If checked, the browswer will automatically open any reader view compatible page in reader view.

**Comment by mthurmond** [comment]
![image](https://user-images.githubusercontent.com/764620/58382670-10cc6580-7f9b-11e9-85e6-1f92b96bfe95.png)

**Comment by PalmerAL** [comment]
I agree it would be nice to have this happen automatically (and the setting itself would be fairly simple to add), but in my experience the algorithm for detecting which pages are readerable isn't very accurate, and tends to show the reader button for a lot of pages that aren't actually articles. If that's the case, I imagine you'd have the opposite problem - lots of pages would open in reader mode, and you'd have to click the button to exit - but perhaps it's more accurate for you?

We should also probably experiment with the [detection algorithm built in to Readability](https://github.com/mozilla/readability/blob/master/Readability-readerable.js), which might be more accurate.

BTW, there's a keyboard shortcut (shift+ctrl+r, or you can change it in the settings) to toggle reader view, which might be a bit faster than clicking the button.

**Comment by Tomadelostacos** [comment]
Maybe have an array of URLs of website @mthurmond often visit in reader mode, and automatically toggle reader mode when on it ?

**Comment by PalmerAL** [comment]
It basically works like that already in 1.10: the second time you open an article from a particular domain in reader mode, you'll get this banner:
<img width="675" src="https://user-images.githubusercontent.com/10314059/58435623-d9d37e00-8086-11e9-96e5-110281dd85f6.png">
and if you choose "yes", future articles from that domain will open in reader mode automatically. So I think this issue is more about automatic reader mode for sites you've never visited before.

**Comment by mthurmond** [comment]
@PalmerAL Thanks for the helpful shortcut and the heads up about the "always open..." feature. I hadn't noticed that before but just realized the prompt pops up once I've opened two or more articles in reader view within a particular URL. 

For now, I think the short-cut plus "always open..." functionality is fine for my use case, though I do think allowing the user to set default reader view preference (i.e. automatically open or manually open) in the future could be worthwhile.

**Comment by PalmerAL** [comment]
As of 9ee8ce1eb7b20efabfb64a5c6a7e03ba39afc7b4, the automatic reader banner will show up on every article, even if you've never opened a page from that site in reader view before. I'll try to look at updating the article-detection algorithm at some point; if we could make that accurate enough, it should be possible to offer a preference for this.
